Assets/Scripts/Features/Gameplay/Camera/CameraDragging.cs

```diff
@@ -0,0 +1,95 @@
+﻿using UnityEngine;
+
+///<summary>
+/// A class that is put on the camera, to control the vertical and horizontal movement of the object.
+/// Uses public float boundaries to show the outer bounds of the field where the object can move.
+///</summary>
+public class CameraDragging : MonoBehaviour
+{
+    [SerializeField]
+    ///<summary>
+    /// dragSpeed determines how fast the camera moves when dragging the screen.
+    ///</summary>
+    public float dragSpeed = 4;
+    ///<summary>
+    /// The camera boundary on the left side.
+    ///</summary>
+    public float leftBoundary = -3f;
+    ///<summary>
+    /// The camera boundary on the right side.
+    ///</summary>
+    public float rightBoundary = 3f;
+    ///<summary>
+    /// The camera boundary on the upper side.
+    ///</summary>
+    public float upBoundary = 3f;
+    ///<summary>
+    /// The camera boundary on the lower side.
+    ///</summary>
+    public float downBoundary = -3f;
+
+    private Vector3 _oldPos;
+    private Vector3 _mouseOrigin;
```

Comment:
This Vector3 is being used to save the mouse position, though a mobile device doesn't normally use one. the mouse position/clicks should be replaced with touch data.